reinforcement_learning/rlclientlib/live_model_impl.cc

```diff
@@ -99,12 +104,20 @@ namespace reinforcement_learning {
     : _configuration(config),
       _error_cb(fn, err_context),
       _data_cb(_handle_model_update, this),
+      _watchdog(&_error_cb),
       _t_factory{t_factory},
       _m_factory{m_factory},
       _logger_factory{logger_factory},
-      _bg_model_proc(config.get_int(name::MODEL_REFRESH_INTERVAL_MS, 60 * 1000), &_error_cb),
-      _buffer_pool(new u::buffer_factory(utility::translate_func('\n', ' ')))
-  {}
+      _bg_model_proc(config.get_int(name::MODEL_REFRESH_INTERVAL_MS, 60 * 1000),_watchdog, "Model downloader", &_error_cb),
+      _buffer_pool(new u::buffer_factory(utility::translate_func('\n', ' '))) {
+    // If there is no user supplied error callback, supply a default one that does nothing but report unhandled background errors.
+    if (fn == nullptr) {
+      _error_cb.set([](const api_status&, void* watchdog_context) {
+        auto watchdog = static_cast<utility::watchdog*>(watchdog_context);
+        watchdog->set_unhandled_background_error(true);
+      }, &_watchdog);
```

Comment:
Is there a reason for this to be a lambda function? Consider making an actual "default_error_callback" function for cleanliness and clarity.

Reply:
An actual function would be much cleaner, agreed!